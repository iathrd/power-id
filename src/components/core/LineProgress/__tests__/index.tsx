import renderer from "react-test-renderer";
import { ThemeProvider } from "@mui/material";
import Component from "../index";
import { mocks } from "@utils/index";

let themeMock = mocks.getThemeMock();

it("renders correctly light mode", () => {
  const tree = renderer
    .create(
      <ThemeProvider theme={themeMock}>
        <Component />
      </ThemeProvider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders correctly light mode", () => {
  themeMock = mocks.getThemeMock("dark");
  const tree = renderer
    .create(
      <ThemeProvider theme={themeMock}>
        <Component />
      </ThemeProvider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
