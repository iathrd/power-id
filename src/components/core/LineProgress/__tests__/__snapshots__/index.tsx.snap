// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly light mode 1`] = `
Array [
  <style
    dangerouslySetInnerHTML={
      Object {
        "__html": ".css-1kxonj9{width:100%;position:relative;}",
      }
    }
    data-emotion="css 1kxonj9"
  />,
  <div
    className="MuiBox-root css-1kxonj9"
  >
    <style
      dangerouslySetInnerHTML={
        Object {
          "__html": ".css-14j5k7k{position:absolute;width:100%;}",
        }
      }
      data-emotion="css 14j5k7k"
    />
    <div
      className="MuiBox-root css-14j5k7k"
    >
      <style
        dangerouslySetInnerHTML={
          Object {
            "__html": ".css-xcrs9u-MuiLinearProgress-root{position:relative;overflow:hidden;display:block;height:4px;z-index:0;background-color:rgb(193, 188, 219);}@media print{.css-xcrs9u-MuiLinearProgress-root{-webkit-print-color-adjust:exact;color-adjust:exact;}}",
          }
        }
        data-emotion="css xcrs9u-MuiLinearProgress-root"
      />
      <span
        aria-valuemax={100}
        aria-valuemin={0}
        aria-valuenow={0}
        className="MuiLinearProgress-root MuiLinearProgress-colorPrimary MuiLinearProgress-determinate css-xcrs9u-MuiLinearProgress-root"
        role="progressbar"
      >
        <style
          dangerouslySetInnerHTML={
            Object {
              "__html": ".css-1l32dws-MuiLinearProgress-bar1{width:100%;position:absolute;left:0;bottom:0;top:0;-webkit-transition:-webkit-transform .4s linear;transition:transform .4s linear;transform-origin:left;background-color:#5e50a1;}",
            }
          }
          data-emotion="css 1l32dws-MuiLinearProgress-bar1"
        />
        <span
          className="MuiLinearProgress-bar MuiLinearProgress-barColorPrimary MuiLinearProgress-bar1Determinate css-1l32dws-MuiLinearProgress-bar1"
          style={
            Object {
              "transform": "translateX(-100%)",
            }
          }
        />
      </span>
    </div>
  </div>,
]
`;

exports[`renders correctly light mode 2`] = `
Array [
  <style
    dangerouslySetInnerHTML={
      Object {
        "__html": ".css-1kxonj9{width:100%;position:relative;}",
      }
    }
    data-emotion="css 1kxonj9"
  />,
  <div
    className="MuiBox-root css-1kxonj9"
  >
    <style
      dangerouslySetInnerHTML={
        Object {
          "__html": ".css-14j5k7k{position:absolute;width:100%;}",
        }
      }
      data-emotion="css 14j5k7k"
    />
    <div
      className="MuiBox-root css-14j5k7k"
    >
      <style
        dangerouslySetInnerHTML={
          Object {
            "__html": ".css-pvvytb-MuiLinearProgress-root{position:relative;overflow:hidden;display:block;height:4px;z-index:0;background-color:rgb(32, 20, 85);}@media print{.css-pvvytb-MuiLinearProgress-root{-webkit-print-color-adjust:exact;color-adjust:exact;}}",
          }
        }
        data-emotion="css pvvytb-MuiLinearProgress-root"
      />
      <span
        aria-valuemax={100}
        aria-valuemin={0}
        aria-valuenow={0}
        className="MuiLinearProgress-root MuiLinearProgress-colorPrimary MuiLinearProgress-determinate css-pvvytb-MuiLinearProgress-root"
        role="progressbar"
      >
        <style
          dangerouslySetInnerHTML={
            Object {
              "__html": ".css-1ir1ncg-MuiLinearProgress-bar1{width:100%;position:absolute;left:0;bottom:0;top:0;-webkit-transition:-webkit-transform .4s linear;transition:transform .4s linear;transform-origin:left;background-color:#4029ab;}",
            }
          }
          data-emotion="css 1ir1ncg-MuiLinearProgress-bar1"
        />
        <span
          className="MuiLinearProgress-bar MuiLinearProgress-barColorPrimary MuiLinearProgress-bar1Determinate css-1ir1ncg-MuiLinearProgress-bar1"
          style={
            Object {
              "transform": "translateX(-100%)",
            }
          }
        />
      </span>
    </div>
  </div>,
]
`;
